src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java

```diff
@@ -1,25 +1,188 @@
+import org.hl7.fhir.dstu3.model.Observation.ObservationStatus;
+import org.hl7.fhir.dstu3.model.Quantity.QuantityComparator;
+import org.hl7.fhir.dstu3.model.SimpleQuantity;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.MissingSegmentException;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
 
-import java.util.ArrayList;
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.List;
 
-import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
 
+@SuppressWarnings("checkstyle:MagicNumber")
+@ExtendWith(MockitoExtension.class)
 class ObservationMapperTest {
+    @Mock
+    private UUIDGenerator uuidGenerator;
+
+    @InjectMocks
+    private ObservationMapper mapper;
+
+    @Test
+    void testMapToObservationsNonePresent() {
+        final Message message = new Message(Collections.emptyList());
+
+        final var observations = mapper.mapToTestGroupsAndResults(message);
+
+        assertThat(observations).isEmpty();
+    }
+
+    @Test
+    void testMapToObservationsMissingRequiredSegments() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "GIS+N" // LabResult
+        ));
+
+        assertThatThrownBy(() -> mapper.mapToTestGroupsAndResults(message))
+            .isExactlyInstanceOf(MissingSegmentException.class)
+            .hasMessageStartingWith("EDIFACT section is missing segment");
+    }
+
+    @Test
+    void testMapToObservationsOnlyRequiredSegmentLaboratoryInvestigation() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "GIS+N", // LabResult
+            "INV+MQ+code:911::description", // LaboratoryInvestigation
+            "GIS+N", // LabResult
+            "INV+MQ+code:911::description" // LaboratoryInvestigation
+        ));
+
+        final var observations = mapper.mapToTestGroupsAndResults(message);
+
+        assertThat(observations).hasSize(2)
+            .allSatisfy(observation -> assertThat(observation.getCode()).satisfies(codeConcept -> assertAll(
+                () -> assertThat(codeConcept.getText()).isEqualTo("description"),
+                () -> assertThat(codeConcept.getCoding()).hasSize(1).first()
+                    .extracting(Coding::getCode)
+                    .isEqualTo("code")
+            )));
```

Comment:
Should we also be testing for the "status" element here? It seems to be required by the FHIR documentation

Reply:
I'll add a test for status becoming `"unknown"` if it's not specified